project-base/storefront/urql/errorExchange.ts

```diff
@@ -77,3 +106,11 @@ const handleCartError = ({ userError, applicationError }: ParsedErrors) => {
         }
     }
 };
+
+const hasFriendlyUrlQueryFailedWith500 = (operation: Operation, error: CombinedError) =>
+    error.graphQLErrors.some((graphqlError) => graphqlError.extensions.code === 500) &&
+    operation.query.definitions.some(
+        (definition) =>
+            definition.kind === Kind.OPERATION_DEFINITION &&
+            definition.directives?.some((directiveDefinition) => directiveDefinition.name.value === 'friendlyUrl'),
+    );
```

Comment:
Maybe you can consider this way. Just idea.

```suggestion
    error.graphQLErrors.some(({extensions}) => extensions.code === 500) &&
    operation.query.definitions.some(
        ({kind, directives}) =>
            kind === Kind.OPERATION_DEFINITION &&
            directives?.some(({name}) => name.value === 'friendlyUrl'),
    );
```

Reply:
I changed the first one, as that was possible, but the second one was kept, as `directives` only exists on it if `kind === Kind.OPERATION_DEFINITION` is true. Because of that, I first need to check it.